network/network.go

```diff
@@ -175,7 +176,7 @@ type network struct {
 // NewNetwork returns a new Network implementation with the provided parameters.
 func NewNetwork(
 	config *Config,
-	minCompatibleTime time.Time,
+	upgradeConfig upgrade.Config,
```

Comment:
This change in the to depend on `upgrade` looks unrelated to the intent of this PR to me... can we remove changes from `network.go` and `network_test.go` from the diff?

Reply:
See [Cam's comment above](https://github.com/ava-labs/avalanchego/pull/3842#issuecomment-2755807516) for the context here. We can do this just by updating `test_network.go` (as we had originally), but that would make it such that multiple places need to be updated for every network upgrade to ensure compatibility, which is more error prone. This approach makes it so that the `minCompatibleTime` only needs to be updated in a single place, but requires then passing the full `upgrade.Config` to `NewNetwork`.

Open to whichever is preferred on your side.

Reply:
From the perspective of `network` it just seems like an abstraction that it doesn't need to know about - `upgrade.Config` defines all fork times but the networking layer only needs to know when a peer isn't compatible.

If the problem we're trying to solve is to make sure that all tests use the same network timestamp, I don't know if this solves our problem because passing in the wrong timestamp seems to be as error prone as passing in the wrong config

Reply:
I switched back to passing the `minCompatibleTime`, which also minimize the diff. We'll just need to remember to update it in `TestNetwork` alongside any bumps for network upgrades made in `node.go`. 🫡